examples/src/walkable-path.ts

```diff
@@ -0,0 +1,151 @@
+  items: [
+    {
+      from: { x: number; y: number; z: number } | { nodeId: number };
+      to: { x: number; y: number; z: number } | { nodeId: number };
+      movingObjectSize: { diameter: number; height: number };
+    }
+  ];
+}
+
+// Should find a way to add this to maybe cognite-sdk-js later?
+async function getTransitPath(client: CogniteClient, transitRequest: TransitPathRequest) {
+  const project = '3ddemo';
+  const url = `https://api.cognitedata.com/api/playground/projects/${project}/3d/pathfinder/transit`;
+  const response = await client.post<TransitPathResponse>(url, {
+    data: transitRequest
+  });
+  return response;
+}
+
+// TODO: Should be added to reveal/viewer. The input data TransitPath response is in my
+// opinion too coupled with our sdk response, should we split it into two funcitons, one taking in a { x, y, z}[]
+// and another converting the response to said array.
+
+function createWalkablePath(pointData: TransitPathResponse, modelTransformation: SectorModelTransformation) {
+  const pathVectors: THREE.Vector3[] = [];
+  const vector: vec3 = vec3.create();
+  for (const item of pointData.items) {
+    for (const segment of item.segments) {
+      for (const path of segment.path) {
+        vec3.set(vector, path.x, path.y, path.z);
+        pathVectors.push(toThreeVector3(new THREE.Vector3(), vector, modelTransformation));
+      }
+    }
+  }
+  return pathVectors;
+}
+
+// TODO: Add to some kind of primitives like class in reveal viewer?
+
+function createPathGeometry(
+  points: THREE.Vector3[],
+  segments: number,
+  radius: number,
+  radiusSegments: number,
+  closed: boolean
+) {
+  const curve = new THREE.CatmullRomCurve3(points);
+  const geometry = new THREE.TubeBufferGeometry(curve, segments, radius, radiusSegments, closed);
+  return geometry;
+}
+
+async function main() {
+  const client: CogniteClient = new CogniteClient({ appId: 'WalkablePath Example' });
+  client.loginWithOAuth({
+    project: '3ddemo'
+  });
+  const modelUrl = new URL(location.href).searchParams.get('model') || '/primitives';
+  const cadModel = await reveal.createLocalCadModel(modelUrl);
+
+  const renderer = new THREE.WebGLRenderer();
+  renderer.setClearColor('#444');
+  renderer.setSize(window.innerWidth, window.innerHeight);
+  document.body.appendChild(renderer.domElement);
+
+  const scene = new THREE.Scene();
+  const cadNode = new CadNode(cadModel);
+  const { position, target, near, far } = cadNode.suggestCameraConfig();
+  const camera = new THREE.PerspectiveCamera(75, window.innerWidth / window.innerHeight, near, far);
+  const controls = new CameraControls(camera, renderer.domElement);
+
+  controls.setLookAt(position.x, position.y, position.z, target.x, target.y, target.z);
+  controls.update(0.0);
+  camera.updateMatrixWorld();
+  scene.add(cadNode);
+
+  const diameter = 0.55;
+  const height = 1.6;
+
+  const walkablePathResponse = await getTransitPath(client, {
+    modelId: 3329507622597457,
```

Comment:
I realize this is problematic as this is very dependent on the SDK, but how could we avoid having the modelId in the code? We should maybe add a task that allows the user to click two objects and create a path (when a model has been loaded from CDF, and not from local storage)?

Reply:
I think we should create a simple feature that we can reuse in other examples for this purpose: A simple box with fields for project + modelId. Then redirect to the proper URL (localhost:8080/example...?project=...&model=...). Bonus points for storing the last result in localStorage or indexedDB :)

Reply:
I have added this but with a default model Id. This way we have the ability to change it. In the future this should be available through the API.

Reply:
created modelId url parameter